model/src/main/proto/topic.proto

```diff
@@ -444,6 +463,12 @@ message UpcomingTopic {
   // The title of the topic.
   SubtitledHtml title = 7;
 
+  // The ID of the classroom this topic is part of.
+  string classroom_id = 8;
+
+  // The title of the classroom this topic is part of.
+  SubtitledHtml classroom_title = 9;
```

Comment:
Here & elsewhere: I'm not sure this will work correctly. SubtitledHtml in activity containers is expected to be localized by the WrittenTranslationContext of that activity's ephemeral object. However, this title belongs to a different context (classroom), so it won't have translations in this activity's WrittenTranslationContext.

You could put the actual classroom title in the ephemeral version, or just rely on multiple fetches from the controller to get the title in cases where we actually need it.

Reply:
I have moved the classroom titles to the ephemeral versions and also added a `classroom_written_translation_context` wherever the `classroom_title` field is present.